spec/game_looper_spec.rb

```diff
@@ -21,71 +21,128 @@ def initialize(marker)
+  def win?
+    win_values.shift
+  end
 
-      allow(outcome_checker).to receive(:draw?).and_return(false, false, false, false, true)
-      allow(outcome_checker).to receive(:win?).and_return(false, false, false, false, false)
+  def draw?
+    draw_values.shift
+  end
+end
 
-      allow(display).to receive(:present).and_return('board', 'board', 'board_with_draw')
+class TestUI
+  attr_reader :output, :input
 
-      game_looper = GameLooper.new(console:, prompt:, board:, display:,
-                                   players:, outcome_checker:)
+  def initialize(input:)
+    @output = []
+    @input = input
+  end
 
-      game_looper.loop
+  def display_board(_board)
+    output << 'board displayed'
+  end
 
-      expected_console_messages = %w[board board board_with_draw]
+  def prompt
+    input.shift
+  end
+end
 
-      expect(console.messages).to eq(expected_console_messages)
+describe 'Game Looper' do
+  describe '#loop' do
+    context 'until there is a draw' do
+      it 'loops' do
+        test_input = []
+        test_win_values = []
+        test_draw_values = []
+        ui = TestUI.new(input: test_input)
+        board = TestBoard.new
+        outcome_checker = TestOutcomeCheckerForGameLooper.new(board:, win_values: test_win_values,
+                                                              draw_values: test_draw_values)
+        player_one = TestPlayer.new('X')
+        player_two = TestPlayer.new('O')
+        players = [player_one, player_two]
+
+        allow(outcome_checker).to receive(:draw?).and_return(false, false, true)
+        allow(outcome_checker).to receive(:win?).and_return(false, false, false)
+
+        game_looper = GameLooper.new(ui:, board:, players:, outcome_checker:)
+
+        game_looper.loop
+
+        expect(ui.output).to eq(['board displayed', 'board displayed', 'board displayed'])
+      end
     end
 
-    it 'loops until there is a win' do
-      console = TestConsole.new
-      prompt = TestPrompt.new
-      board = TestBoard.new
-      display = TestDisplay.new
-      outcome_checker = TestOutcomeCheckerForGameLooper.new(board:)
-      player_one = TestPlayer.new('X')
-      player_two = TestPlayer.new('O')
-      players = [player_one, player_two]
+    context 'until there is a win' do
+      it 'loops' do
+        test_input = []
+        ui = TestUI.new(input: test_input)
+        board = TestBoard.new
+        win_values = [false, false, true]
+        draw_values = [false, false, false]
+        outcome_checker = TestOutcomeCheckerForGameLooper.new(board:, win_values:, draw_values:)
+        player_one = TestPlayer.new('X')
+        player_two = TestPlayer.new('O')
+        players = [player_one, player_two]
 
-      allow(outcome_checker).to receive(:draw?).and_return(false, false, false, false, false)
-      allow(outcome_checker).to receive(:win?).and_return(false, false, false, false, true)
+        allow(outcome_checker).to receive(:draw?).and_return(false, false, false)
```

Comment:
Same question as above about these `outcome_checker` values.

Reply:
I got halfway through removing the rspec-mock "allow" methods and forgot to finish - thanks for catching that!